presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -520,6 +520,7 @@ public void fail(Throwable cause)
         requireNonNull(cause, "cause is null");
 
         stateMachine.transitionToFailed(cause);
+        stateMachine.updateQueryInfo(Optional.empty());
```

Comment:
Haven't looked in detail yet, but I think we might want:
```java 
if (stateMachine.transitionToFailed(cause)) {
    stateMachine.updateQueryInfo(Optional.empty());
}
```

Reply:
Fixed. Also isolated the test change I made. Hopefully tests will pass.